compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/CupertinoOverscrollEffect.kt

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation
+
+import androidx.compose.animation.core.AnimationState
+import androidx.compose.animation.core.VectorConverter
+import androidx.compose.animation.core.animateTo
+import androidx.compose.animation.core.spring
+import androidx.compose.foundation.gestures.FlingIntoOverscrollEffect
+import androidx.compose.foundation.gestures.ScrollValueConverter
+import androidx.compose.foundation.layout.offset
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.input.nestedscroll.NestedScrollSource
+import androidx.compose.ui.layout.onPlaced
+import androidx.compose.ui.unit.*
+import kotlin.math.abs
+import kotlin.math.sign
+
+enum class CupertinoScrollSource {
+    DRAG, FLING
+}
+
+enum class CupertionOverscrollOffsetSpace {
+    LINEAR,
+    RUBBER_BANDED
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+class CupertinoOverscrollEffect : OverscrollEffect {
```

Comment:
Should we move `Cupertino*` to separate package?

Reply:
I think it's fine for now, especially considering they are in different packages (foundation/animation), same as the classes they are based on.